kombu/connection.py

```diff
@@ -169,7 +169,8 @@ def __init__(self, hostname='localhost', userid=None,
             if ';' in hostname:
                 alt.extend(hostname.split(';'))
                 hostname = alt[0]
-            if '+' in hostname[:hostname.index('://')]:
+            scheme = hostname[:hostname.index('://')]
+            if 'mongodb+srv' not in scheme and '+' in scheme:
```

Comment:
wonder if instead adding `mongodb+srv` explicitly here, it wouldn't make sense to check if scheme is in `TRANSPORT_ALIASES` (or better if you can get a transport for it) before splitting by `+`.

If you can't get a transport for the whole scheme, then you split by `+` and check if you can get a transport for the part before `+`.

That would allow to support any transport that has `+` in the scheme without having to explicitly list them here.